modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/query/Sort.scala

```diff
@@ -0,0 +1,93 @@
+package ch.epfl.bluebrain.nexus.kg.indexing.query
+
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.kg.indexing.query.Sort.OrderType
+import ch.epfl.bluebrain.nexus.kg.indexing.query.Sort.OrderType._
+import ch.epfl.bluebrain.nexus.kg.indexing.query.SortList._
+
+import scala.util.Try
+
+/**
+  * Data type of a collection of [[Sort]]
+  * @param values the values to be sorted in the provided order
+  */
+final case class SortList(values: List[Sort]) extends Product with Serializable {
+  private lazy val toVarsMapping = values.zipWithIndex.map { case (sort, i) => sort -> s"?sort$i" }
+
+  /**
+    * @return the string representation of the variables used for sorting (inside the SELECT clause) in SPARQL language
+    */
+  def toVars = toVarsMapping map {case (_, variable) => variable} mkString (" ")
+
+  /**
+    * @return the string representation of the optional triples used for sorting (inside the WHERE clause) in SPARQL language
+    */
+  def toTriples =
+    toVarsMapping map {
+      case (Sort(_, predicate), variable) => s"?s <${predicate}> $variable"
+    } mkString ("}\nOPTIONAL{") prefixSuffixNonEmpty ("OPTIONAL{", "}")
```

Comment:
I think we should not overuse infix notations. For something like `mkString` it's not very natural.
http://docs.scala-lang.org/style/method-invocation.html#infix-notation